sunkit_image/stara.py

```diff
@@ -0,0 +1,154 @@
+        `skimage.morphology.white_tophat`. This is the maximum radius of
+        detected features.
+
+    median_box : `astropy.units.Quantity`, optional
+        The size of the structuring element for the median filter, features
+        smaller than this will be averaged out.
+
+    threshold : `int`, optional
+        The threshold used for detection, this will be subject to detector
+        degradation. The default is a reasonable value for HMI continuum images.
+
+    limb_filter : `astropy.units.Quantity`, optional
+        If set, ignore features close to the limb within a percentage of the
+        radius of the disk. A value of 10% generally filters out false
+        detections around the limb with HMI continuum images.
+    """
+    data = invert(smap.data)
+
+    # Filter things that are close to limb to reduce false detections
+    if limb_filter is not None:
+        hpc_coords = sunpy.map.all_coordinates_from_map(smap)
+        r = np.sqrt(hpc_coords.Tx**2 + hpc_coords.Ty**2) / (smap.rsun_obs - smap.rsun_obs * limb_filter)
+        data[r > 1] = np.nan
+
+    # Median filter to remove detections based on hot pixels
+    m_pix = int((median_box / smap.scale[0]).to_value(u.pix))
+    med = median(data, square(m_pix), behavior="ndimage")
+
+    # Construct the pixel structuring element
+    c_pix = int((circle_radius / smap.scale[0]).to_value(u.pix))
+    circle = disk(c_pix / 2)
+
+    finite = white_tophat(med, circle)
+    finite[np.isnan(finite)] = 0  # Filter out nans
+
+    return finite > threshold
+
+
+def get_regions(segmentation, smap):
+    """
+    Extracts regions from a segmented image and returns a table with their
+    properties.
+
+    Parameters
+    ----------
+    segmentation : `numpy.ndarray`
+        A 2D array representing the segmented image, where different regions are marked with different integer labels.
+
+    smap : `sunpy.map.GenericMap`
+        The original SunPy map from which the segmented image was derived. This is used to convert pixel coordinates to world coordinates.
+
+    Returns
+    -------
+    regions : `astropy.table.QTable`
+        A table containing the properties of each detected region. The properties include the label, centroid, area, and minimum intensity of each region, as well as the observation time and the heliographic Stonyhurst coordinates of the region center. If no regions are detected, an empty table is returned.
+    """
+    labelled = label(segmentation)
+    if labelled.max() == 0:
+        return QTable()
+
+    regions = regionprops_table(labelled, smap.data, properties=["label", "centroid", "area", "min_intensity"])
+
+    regions["obstime"] = Time([smap.date] * regions["label"].size)
+    regions["center_coord"] = smap.pixel_to_world(
+        regions["centroid-0"] * u.pix,
+        regions["centroid-1"] * u.pix,
+    ).heliographic_stonyhurst
+
+    return QTable(regions)
+
+
+def plot_sunspots(segs, maps):
```

Comment:
This function has to go but the code should be used in a new gallery example for this code.

Reply:
Sure will add a gallery example.

Reply:
Should I add the example in this PR itself or make a new one?

Reply:
In this PR please.